src/core_atmosphere/dynamics/mpas_atm_boundaries.F

```diff
@@ -176,26 +180,86 @@ subroutine mpas_atm_update_bdy_tend(clock, streamManager, block, firstCall, ierr
         call mpas_pool_get_array(lbc, 'lbc_scalars', scalars, 2)
 
         call mpas_pool_get_array(mesh, 'cellsOnEdge', cellsOnEdge)
-        call mpas_pool_get_dimension(mesh, 'nCells', nCells)
-        call mpas_pool_get_dimension(mesh, 'nEdges', nEdges)
-        call mpas_pool_get_dimension(lbc, 'index_qv', index_qv)
+        call mpas_pool_get_dimension(mesh, 'nCells', nCells_ptr)
+        call mpas_pool_get_dimension(mesh, 'nEdges', nEdges_ptr)
+        call mpas_pool_get_dimension(mesh, 'nVertLevels', nVertLevels_ptr)
+        call mpas_pool_get_dimension(state, 'num_scalars', nScalars_ptr)
+        call mpas_pool_get_dimension(lbc, 'index_qv', index_qv_ptr)
         call mpas_pool_get_array(mesh, 'zz', zz)
 
+        MPAS_ACC_TIMER_START('mpas_atm_update_bdy_tend [ACC_data_xfer]')
+        if (.not. firstCall) then
+            call mpas_pool_get_array(lbc, 'lbc_u', lbc_tend_u, 1)
+            call mpas_pool_get_array(lbc, 'lbc_ru', lbc_tend_ru, 1)
+            call mpas_pool_get_array(lbc, 'lbc_rho_edge', lbc_tend_rho_edge, 1)
+            call mpas_pool_get_array(lbc, 'lbc_w', lbc_tend_w, 1)
+            call mpas_pool_get_array(lbc, 'lbc_theta', lbc_tend_theta, 1)
+            call mpas_pool_get_array(lbc, 'lbc_rtheta_m', lbc_tend_rtheta_m, 1)
+            call mpas_pool_get_array(lbc, 'lbc_rho_zz', lbc_tend_rho_zz, 1)
+            call mpas_pool_get_array(lbc, 'lbc_rho', lbc_tend_rho, 1)
+            call mpas_pool_get_array(lbc, 'lbc_scalars', lbc_tend_scalars, 1)
```

Comment:
Since we now get pointers to arrays here, I think it would be fine to include in this PR the removal of redundant calls to `mpas_pool_get_array` below (beginning around line 270).